fabric_content/data_engineering/example_spark_pipeline/spark_job/tests/end_to_end/steps/test_transform_and_save.py

```diff
@@ -0,0 +1,191 @@
+GOLD_LAKEHOUSE_ID = require_env("GOLD_LAKEHOUSE_ID")
+FABRIC_CLIENT_ID = require_env("FABRIC_CLIENT_ID")
+FABRIC_CLIENT_SECRET = require_env("FABRIC_CLIENT_SECRET")
+FABRIC_TENANT_ID = require_env("FABRIC_TENANT_ID")
+
+
+@pytest.fixture(scope="module")
+def spark():
+    """Fixture to provide a SparkSession for the test module and clean up after."""
+    spark = (
+        SparkSession.builder
+            .appName("Testing PySpark Example")
+            .config("spark.driver.bindAddress", "127.0.0.1")
+            .config("spark.driver.host", "127.0.0.1")
+            .config(
+                "spark.jars.packages",
+                "io.delta:delta-spark_2.12:3.1.0," \
+                "org.apache.hadoop:hadoop-azure:3.4.1," \
+                "org.apache.hadoop:hadoop-common:3.4.1," \
+                "com.azure:azure-storage-blob:12.30.0"
+            )
+            .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+            .config("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog")
+            .config("fs.azure.account.auth.type", "OAuth")
+            .config("fs.azure.account.oauth.provider.type", "org.apache.hadoop.fs.azurebfs.oauth2.ClientCredsTokenProvider")
+            .config("fs.azure.account.oauth2.client.id", FABRIC_CLIENT_ID)
+            .config("fs.azure.account.oauth2.client.secret", FABRIC_CLIENT_SECRET)
+            .config("fs.azure.account.oauth2.client.endpoint", f"https://login.microsoftonline.com/{FABRIC_TENANT_ID}/oauth2/token")
+            .config("spark.jars.repositories", "https://repo1.maven.org/maven2,https://mvnrepository.com/artifact")
+            .getOrCreate()
+    )
+    yield spark
+    spark.stop()
+
+
+@pytest.fixture
+def fabric_helper():
+    """Fixture to provide a FabricHelper instance for the test."""
+    return FabricHelper(
+        tenant_id=FABRIC_TENANT_ID,
+        client_id=FABRIC_CLIENT_ID,
+        client_secret=FABRIC_CLIENT_SECRET
+    )
+
+
+def get_workspace_id_by_name(workspace_name: str) -> str:
+    """Map workspace names to their corresponding IDs.
+
+    Args:
+        workspace_name (str): The logical name of the workspace (e.g., 'engineering', 'gold').
+
+    Returns:
+        str: The corresponding workspace ID.
+
+    Raises:
+        ValueError: If the workspace name is not recognized.
+    """
+    workspace_ids = {
+        "engineering": ENGINEERING_WORKSPACE_ID,
+        "gold": GOLD_WORKSPACE_ID,
+    }
+    if workspace_name not in workspace_ids:
+        raise ValueError(f"Unknown workspace name: {workspace_name}")
+    return workspace_ids[workspace_name]
+
+
+def get_pipeline_id_by_name(items: list[dict], pipeline_name: str) -> str:
+    """Find the pipeline ID by display name from the workspace items list.
+
+    Args:
+        items (list[dict]): List of workspace items from the Fabric API.
+        pipeline_name (str): The display name of the pipeline to find.
```

Comment:
```suggestion
        items: List of workspace items from the Fabric API.
        pipeline_name: The display name of the pipeline to find.
```

they are already in type hints - no need to repeat